src/Svg.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import './Svg.css';
+
+export class Svg extends Component {
+    static getDerivedStateFromProps(props, state) {
+        return {
+            contents: typeof props.svg === 'string' ? props.svg : state.contents,
+        };
+    }
+
+    promise = undefined;
+    state = { contents: '' };
+
+    componentDidMount() {
+        this.maybeWaitForSvg();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.svg !== prevProps.svg) {
+            this.maybeWaitForSvg();
+        }
+    }
+
+    componentWillUnmount() {
+        this.resetWaitForSvg();
+    }
+
+    render() {
+        const { svg, className, forwardedRef, ...rest } = this.props;
+        const { contents } = this.state;
+        const finalProps = {
+            ...rest,
+            className: classNames('svgWrapper', className),
+        };
+
+        return <i { ...finalProps } ref={ forwardedRef } dangerouslySetInnerHTML={ { __html: contents } } />;
+    }
+
+    resetWaitForSvg() {
+        this.promise = undefined;
+    }
+
+    async maybeWaitForSvg() {
+        const { svg } = this.props;
+
+        this.resetWaitForSvg();
+
+        if (typeof svg !== 'object') {
+            return;
+        }
+
+        const promise = this.promise = svg;
+        const result = await this.promise.catch(() => ({ default: '' }));
+
+        /* istanbul ignore next */
```

Comment:
Why is this next line not tested?

Reply:
Couldn't find a swift way to get coverage on this line, so I decided to ignore it to get the coverage to get to 100% (it was at about 90% when uncommented). Decided to do it because the time getting the coverage would be more well spent on other more important tasks, but I could uncomment the line altogether, if it's best?

Reply:
Isn't it as simple as the SVG being equal to `new Promise((resolve) => resolve({ default: 'batatas'}))` ?

Reply:
I don't think it is. It's a matter of an "Else path not taken", tricky situation to test due to this condition involving two promises that shouldn't be equal. Making them not equal is the actual part I struggled with.

Reply:
I just realize this line doesn't make much sense. I don't see a situation where `promise` would be different from `this.promise`

Reply:
@ruimonteiro93 tomorrow I can try and help on how to cover this section.